### PR TITLE
[SQL Server] (No.4) add sql server creation codes

### PR DESCRIPTION
**Pull request by shenqianjin**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/sqlserver/creation/SqlServerCreationDialog.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import com.microsoft.azure.toolkit.intellij.common.AzureDialog;
+import com.microsoft.azure.toolkit.lib.common.form.AzureForm;
+import com.microsoft.azure.toolkit.lib.sqlserver.SqlServerConfig;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+
+import static com.microsoft.intellij.ui.messages.AzureBundle.message;
+
+public class SqlServerCreationDialog extends AzureDialog<SqlServerConfig> {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/sqlserver/creation/SqlServerCreationDialog.java:35]: DialogWrapper

Comment (shenqianjin) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/sqlserver/creation/SqlServerCreationDialog.java:35]: why? these are a few custom functions lie in AzureDialog.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/sqlserver/CreateSqlServerAction.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        dialog.setOkActionListener((data) -> this.createSqlServer(data, dialog));
+        dialog.show();
+    }
+
+    private void createSqlServer(final SqlServerConfig config, SqlServerCreationDialog dialog) {
+        final Runnable runnable = () -> {
+            final ProgressIndicator indicator = ProgressManager.getInstance().getProgressIndicator();
+            indicator.setIndeterminate(true);
+            DefaultLoader.getIdeHelper().invokeLater(dialog::close);
+            ISqlServer server = SqlServerService.getInstance().create(config);
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/sqlserver/CreateSqlServerAction.java:77]: SqlServerService is a bit confusing

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/sqlserver/creation/SqlServerCreationDialog.java (hunk 1 above)
Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/sqlserver/creation/SqlServerCreationDialog.java:35]: do you know the difference between AzureDialogWrapper and AzureDialog?

Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/sqlserver/creation/SqlServerCreationDialog.java:35]: We will deprecate AzureDialogWrapper,  use AzureDialog is sugguested.

Comment (shenqianjin) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/sqlserver/creation/SqlServerCreationDialog.java:35]: It is -- AzureDialog.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/sqlserver/CreateSqlServerAction.java (hunk 2 above)
Comment (shenqianjin) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/sqlserver/CreateSqlServerAction.java:77]: the class is not in this PR. Will rename it to CreateSqlServerTask in next PR.

Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/sqlserver/CreateSqlServerAction.java:77]: keep this comment unsolved until new PR is merged

**Review by andxu — APPROVED** [review]

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/sqlserver/CreateSqlServerAction.java (hunk 2 above)
Comment (shenqianjin) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/sqlserver/CreateSqlServerAction.java:77]: updated in https://github.com/microsoft/azure-tools-for-java/commit/ce6109b33be79c9d8552a3d1fedd83c86531e7bb

**Pull request merged**